src/components/schedule-page/table/schedule/schedule-parts/error-tooltip-provider.component.tsx

```diff
@@ -36,37 +37,47 @@ export function ErrorTooltipProvider({
   const errors = useSelector((state: ApplicationStateModel) =>
     errorSelector(state.actualState.scheduleErrors)
   );
+  const dispatch = useDispatch();
+
+  const handleShow = (): void => {
+    dispatch(ScheduleDataActionCreator.hideErrors());
+  };
   let triangleStyle = "single";
   errors.forEach((element) => {
     triangleStyle = element["className"] || triangleStyle;
   });
-
   const errorTriangle = useRef<HTMLDivElement>(null);
   const container = useRef<HTMLDivElement>(null);
 
   const tooltipRef = useRef<HTMLDivElement>(null);
   const [isToolTipOpen, setToolTipOpen] = useState(false);
+  let isOpen = isToolTipOpen;
+  const manuallySelectedErrors = errors.filter((error) => error.isVisible);
+  if (manuallySelectedErrors.length !== 0)
+    isOpen =
+      manuallySelectedErrors.some((e) => e["className"] === "right") ||
+      manuallySelectedErrors.every((e) => e["className"] === undefined);
```

Comment:
Lepiej zmienić kolejność tych predykatów: najpierw sprawdzać czy jest prawdą to, że wszystkie błędy posiadają `classname`, a potem jeśli są takie które posiadają takie pole sprawdzać czy jest błąd z polem `right`.

Bo teraz to wygląda tak, że:
1. My przechodzimy przez całą listę w `some`
2. Jeśli ona zwraca `false` (co jest możliwe jeśli np. żaden z elementów nie posiada pola `classname`, przechodzimy przez całą listę jeszcze raz w `every`.

Natomiast jeśli my zmienimy koljeność to unikniemy sytuacji o której piszę: żaden z elementów z elementów nie posiada className i my przechodzimy dwa razy przez listę, ponieważ wtedy pierwszy warunek zwróci `false` i my nie wejdziemy w drugi
![image](https://user-images.githubusercontent.com/21079319/113431944-cf9a0d80-93dc-11eb-8904-09520e8bdf2c.png)